src/app/exam/problems/ExamProblems.js

```diff
@@ -0,0 +1,174 @@
+import { useState, useEffect } from "react";
+import {Redirect, useParams, useRouteMatch} from "react-router-dom";
+import { Map } from "immutable"
+
+import { examService, problemService, submissionService } from "../../../services/services.js";
+import { ThreeDotsButton } from "../../commons/buttons/ThreeDotsButton.js";
+import FakeLink from "../../commons/FakeLink.js";
+import { ItemListPage } from "../../commons/ItemListPage/ItemListPage.js";
+import { ExamInPageNavigationBar } from "../ExamInPageNavigationBar";
+import { AddProblemModal } from "../modals/AddProblemModal.js";
+import {RejudgeProblemModal} from "../modals/RejudgeProblemModal";
+import "./ExamProblems.scss";
+
+
+const toCharacterIndex = i => {
+    return String.fromCharCode(i + 65);
+}
+
+const ExamProblems = ({ exams }) => {
+    const { url: currentURL } = useRouteMatch()
+    const { examId } = useParams()
+    const currentExamName = exams.find(exam => exam.id === parseInt(examId))?.name
+    const [examProblems, setExamProblems] = useState(null);
+
+    const [showAddProblemModal, setShowAddProblemModal] = useState(false);
+    const [showRejudgeProblemModal, setShowRejudgeProblemModal] = useState(0);
+    const [problemId2Title, setProblemId2Title] = useState(new Map())
+    const [redirectURL, setRedirectURL] = useState()
+    const [rejudgeProblemId, setRejudgeProblemId] = useState(0)
+
+    const fetchExam = () => {
+        examService.getExam(examId)
+            .then(exam => {
+                exam.questions.sort((questionA, questionB) => questionA.questionOrder - questionB.questionOrder);
+                setExamProblems(exam.questions);
+            });
+    }
+
+    useEffect(() => {
+        setRedirectURL(null)
+        if (!examProblems) {
+            examService.getExam(examId)
+                .then(exam => {
+                    exam.questions.sort((questionA, questionB) => questionA.questionOrder - questionB.questionOrder);
+                    setExamProblems(exam.questions);
+                });
+        } else {
+            examProblems.forEach(problem => {
+                problemService.getProblemById(problem.problemId)
+                    .then(res => setProblemId2Title(prev => prev.set(problem.problemId, res.title)))
+            })
```

Comment:
don't iterate on every problem

the get exam overview api has already included the problem's title